kategory-docs/docs/docs/patterns/glossary/README.md

```diff
@@ -0,0 +1,145 @@
+
+### Typeclasses
+
+A typeclass is an interface representing one behavior associated with a type.
+Examples of this behavior are comparison ([`Eq`]({{ '/docs/typeclasses/eq' | relative_url }})),
+composability ([`Monoid`]({{ '/docs/typeclasses/monoid' | relative_url }})),
+its contents are mappable ([`Functor`]({{ '/docs/typeclasses/functor' | relative_url }})),
+or error recovery ([`MonadError`]({{ '/docs/typeclasses/monaderror' | relative_url }})).
+
+```kotlin
+interface Eq<F>: Typeclass {
+  fun eqv(a: F, b: F): Boolean
+}
+```
+
+What differentiates typeclasses from regular interfaces is that they are meant to be created at a global scope for a single type.
+The association is done using generic parametrization rather than the usual subclassing. This means that they can be implemented for any class, even those not in the current project.
+
+### Instances
+
+A single implementation of a typeclass for a specific datatype or class.
+Because typeclasses require generic parameters each implementation is meant to be unique for that parameter.
+
+```kotlin
+@instance
+object IntEqInstance: Eq<Int> {
+  override fun eqv(a: Int, b: Int): Boolean = a == b
+}
+```
+
+In KΛTEGORY all typeclass instances can be looked up using a method with the same name as the typeclass.
+All the instances in the library are already registered and available in the global scope.
+If you're defining your own instances and would like for them to be discoverable in the global scope 
+you can add them by annotating them as `@instance`, and KΛTEGORY's [annotation processor](https://github.com/kategory/kategory#additional-setup) will register them for you.
+
+```kotlin:ank
+import kategory.*
+
+eq<Int>()
+```
+
+### Type constructors
+
+> NOTE: This approach to type constructors will be simplified if [KEEP-87](https://github.com/Kotlin/KEEP/pull/87) is approved. Go vote!
+
+A type constructor is any class or interface that has at least one generic parameter. For example, 
+`ListKW<A>` or `Option<A>`. It's called a constructor because it is similar to a function where the parameter is `A`.
+For example, applying `Int` to the type constructor `ListKW<A>` returns a `ListKW<Int>`.
+This list isn't parametrized in any generic value, so it cannot be considered a type constructor anymore.
+
+Type constructors are useful when matched with typeclasses because they help us represent non-parametrized values.
+As type constructors is not a first class feature in Kotlin, we use an interface `Hk<F, A>` to represent them.
+Hk stands for Higher Kind, which is the name of the feature that allows working with type constuctors as first class.
+
+#### Higher Kinds
+
+In a Higher Kind with the shape `HK<F, A>`, if `A` is the type of the content then `F` has to be the type of the container.
+A malformed container would use the whole type constructor, duplicating the type ~~HK<Option<A>, A>~~.
```

Comment:
is that ~~ properly applied ? It's showing as strikethrough on github: 
<img width="223" alt="captura de pantalla 2017-11-26 a las 11 20 19" src="https://user-images.githubusercontent.com/6547526/33239133-eeb548f2-d29b-11e7-8d64-352873e83e2d.png">

Reply:
Arg